src/web/statics.rs

```diff
@@ -272,23 +280,22 @@ mod tests {
 
     #[test]
     fn directory_traversal() {
-        wrapper(|env| {
-            let web = env.frontend();
-
-            let urls = &[
-                "../LICENSE.txt",
-                "%2e%2e%2fLICENSE.txt",
-                "%2e%2e/LICENSE.txt",
-                "..%2fLICENSE.txt",
-                "%2e%2e%5cLICENSE.txt",
-            ];
-
-            for url in urls {
-                let req = web.get(&format!("/-/static/{}", url)).send()?;
-                assert_eq!(req.status().as_u16(), 404);
-            }
-
-            Ok(())
-        });
+        const PATHS: &[&str] = &[
+            "../LICENSE",
+            "%2e%2e%2fLICENSE",
+            "%2e%2e/LICENSE",
+            "..%2fLICENSE",
+            "%2e%2e%5cLICENSE",
+        ];
+
+        for path in PATHS {
+            // This doesn't test an actual web request as the web framework used at the time of
+            // writing this test (iron 0.5) already resolves `..` before calling any handler.
+            //
+            // Still, the test ensures the underlying function called by the request handler to
+            // serve the file also includes protection for path traversal, in the event we switch
+            // to a framework that doesn't include builtin protection in the future.
+            assert!(serve_file(path).is_err(), "{} was served", path);
```

Comment:
Can you check that these are 404s instead of internal errors?

Reply:
Pushed a test!